lib/avatarService.ts

```diff
@@ -28,14 +29,21 @@ export class AvatarService {
   }
 
   /**
-   * Returns promise of the Avatar TruffleContract
+   * Returns promise of the Avatar TruffleContract.
+   * Returns undefined if not found.
    */
   public async getAvatar(): Promise<any> {
     if (!this.avatar) {
       const Avatar = await Utils.requireContract("Avatar");
-      this.avatar = await Avatar.at(this.avatarAddress);
+      return Avatar.at(this.avatarAddress)
+        .then((avatar: any) => avatar) // only way to get to catch
```

Comment:
@dev-matan-tsuberi 

The way truffle fakes promises there is no `catch` method unless you call `then` first.  `then` returns the true promise.

Reply:
fake promises? is it documented somewhere?

Reply:
They don't call them "fake", and they don't tell you that what they have done is only a partial implementation of a promise.  All they say is that they've provided promise-like functionality to encourage us to get away from callbacks and start using async/await, claiming that what they've done works great with async/await.  But aside from the fact that they only provide a `then` function, its worse:  What they've done causes "unhandled promises" to happen when you try to catch errors using try/catch.  You can't catch them because you haven't awaited the actual promise.  The actual promise is only returned by `then`.  (This is what I _believe_ is happening, haven't absolutely confirmed the hypothesis.)